verifiers/risc0/src/lib.rs

```diff
@@ -32,25 +34,41 @@ pub use fake_extend_benchmarking as extend_benchmarking;
 
 mod verifier_should;
 mod weight;
+mod weight_verify_proof;
+
+pub use crate::weight_verify_proof::WeightInfo as WeightInfoVerifyProof;
+pub use weight::WeightInfo;
 
 pub trait Config: 'static {
-    /// Maximum number of bytes contained in the proof (otherwise rejected)
-    type MaxProofSize: Get<u32>;
+    /// Maximum number of 2^20 segments proof in a composite -> this, combined with
+    /// `Segment20MaxSize`, defines also the maximum proof size (proof bigger than
+    /// this value will be rejected).
+    type MaxNSegment: Get<u32>;
+
+    /// Maximum number of bytes contained in a 2^20 segment size
+    type Segment20MaxSize: Get<u32>;
+
     /// Maximum number of bytes contained in the public inputs (otherwise rejected)
     type MaxPubsSize: Get<u32>;
 
+    /// Weight info used to compute the verify proof weight
+    type WeightInfo: WeightInfoVerifyProof;
+
     fn max_proof_size() -> u32 {
-        Self::MaxProofSize::get()
+        Self::MaxNSegment::get() * Self::Segment20MaxSize::get()
     }
 
     fn max_pubs_size() -> u32 {
         Self::MaxPubsSize::get()
     }
+
+    fn max_verify_proof_weight() -> Weight {
+        Self::WeightInfo::verify_proof_segment_poseidon2_16() * Self::MaxNSegment::get() as u64
```

Comment:
Shouldn't we use `verify_proof_segment_poseidon2_20()`?
```suggestion
        Self::WeightInfo::verify_proof_segment_poseidon2_20() * Self::MaxNSegment::get() as u64
```

Reply:
Yes, you're right... but the real question here is ... why there isn't any tests that catch it?